reana_server/utils.py

```diff
@@ -130,10 +132,57 @@ def remove_fetched_workflows_dir(tmpdir: str) -> None:
         shutil.rmtree(tmpdir)
 
 
-def mv_workflow_files(source: str, target: str) -> None:
-    """Move files from one directory to another."""
-    for entry in os.listdir(source):
-        shutil.move(os.path.join(source, entry), target)
+def mv_workflow_files(source: str, target: str, reana_spec: Dict) -> None:
+    """Move workflow's input files from one directory to another.
+
+    :param source: Directory from which the files will be moved.
+    :param target: Directory where the files will be moved to.
+    :param reana_spec: REANA specification used to decide which files to move.
+    """
```

Comment:
In addition to moving files, this function also validates them. Maybe call it `validate_and_move_workflow_files`?